src/script/cpp_api/s_env.cpp

```diff
@@ -53,13 +53,7 @@ void ScriptApiEnv::environment_Step(float dtime)
 	lua_getfield(L, -1, "registered_globalsteps");
 	// Call callbacks
 	lua_pushnumber(L, dtime);
-	try {
-		runCallbacks(1, RUN_CALLBACKS_MODE_FIRST);
-	} catch (LuaError &e) {
-		getServer()->setAsyncFatalError(
-				std::string("environment_Step: ") + e.what() + "\n"
-				+ script_get_backtrace(L));
-	}
+	runCallbacks(1, RUN_CALLBACKS_MODE_FIRST);
```

Comment:
The server *does* have a central place to catch `LuaError`s, also `runCallbacks` already adds a stacktrace on error.

Reply:
Uh well.. It does abort.
```C++
2021-09-10 19:45:52: WARNING[Server]: Undeclared global variable "erro" accessed at lab/init_107_invalid_returns.lua:2
2021-09-10 19:45:52: ERROR[Server]: An unhandled exception occurred: Runtime error from mod 'lab' in callback environment_Step(): lab/init_107_invalid_returns.lua:2: attempt to call global 'erro' (a nil value)                                                                              
2021-09-10 19:45:52: ERROR[Server]: stack traceback:                                                                                                         
2021-09-10 19:45:52: ERROR[Server]:     (irrelevant)                                                                                                                                                  
2021-09-10 19:45:52: ERROR[Server]: In thread 7fffd37fe700:                                                                                                  
2021-09-10 19:45:52: ERROR[Server]: /data/Minetest/minetest/src/server.cpp:7c: virtual void* ServerThread::run(): A fatal error occurred: Runtime error from mod 'lab' in callback environment_Step(): lab/init_107_invalid_returns.lua:2: attempt to call global 'erro'                       
2021-09-10 19:45:52: ERROR[Server]: (a nil value)                                                                                                            
2021-09-10 19:45:52: ERROR[Server]: stack traceback:                                                                                                         
2021-09-10 19:45:52: ERROR[Server]:     (irrelevant)
--Type <RET> for more, q to quit, c to continue without paging--

Thread 13 "Server" received signal SIGABRT, Aborted.
[Switching to Thread 0x7fffd37fe700 (LWP 12577)]
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50      ../sysdeps/unix/sysv/linux/raise.c: Datei oder Verzeichnis nicht gefunden.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff6e6d859 in __GI_abort () at abort.c:79
#2  0x0000555555aad34c in fatal_error_fn(char const*, char const*, unsigned int, char const*) ()
#3  0x00005555556294da in ServerThread::run() [clone .cold] ()
#4  0x00005555559d2727 in Thread::threadProc(Thread*) ()
#5  0x00007ffff727cde4 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff75d3609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#7  0x00007ffff6f6a293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

Sample:
```Lua
minetest.register_globalstep(function(dtime)
	erro("test")
end)
```

I suspect the same would happen if I triggered an `on_liquid_transformed` callback error

Reply:
Weird, works for me:
<pre>
2021-09-10 20:28:28: ERROR[Main]: ServerError: AsyncErr: Lua: Runtime error from mod 'test2' in callback environment_Step(): /home/stefan/minetest/bin/../mods/test2/init.lua:292: test
2021-09-10 20:28:28: ERROR[Main]: stack traceback:
2021-09-10 20:28:28: ERROR[Main]: 	[C]: in function 'error'
2021-09-10 20:28:28: ERROR[Main]: 	[...]
</pre>

The catch handler is here (even before this PR)
https://github.com/minetest/minetest/blob/bbfae0cc673d3abdc21224c53e09b209ee4688a2/src/server.cpp#L119-L122
and encompasses everything the server does, including the environment step:
`ServerThread::run` -> `Server::AsyncRunStep` -> `ServerEnvironment::step` -> `ScriptApiEnv::environment_Step`

Reply:
(Now fixed)